PowerEditor/src/uchardet/nsSBCharSetProber.h

```diff
@@ -84,7 +84,7 @@ class nsSingleByteCharSetProber : public nsCharSetProber{
   // prober has a hard-coded call to FilterWithoutEnglishLetters which gets rid
   // of the English letters.
   PRBool KeepEnglishLetters() {return mModel->keepEnglishLetter;} // (not implemented)
-  nsSingleByteCharSetProber operator=(const nsSingleByteCharSetProber&){};
+  nsSingleByteCharSetProber operator=(const nsSingleByteCharSetProber&) = delete;

```

Comment:
This is not proper assignment operator overloading. It should perform deep copy and should return ```*this```. 
Previous code (assignment operator overloading) does nothing. So instead of doing nothing, it should not be permitted at all.
There two ways to do it. 1. put = overloading in private section (similar to singleton class). Modern C++ offer another way mark it deleted (which is far better as compiler produce error when it see the usage of = in the code. Refer [msdn](https://msdn.microsoft.com/en-us/library/dn457344.aspx).).